### PR TITLE
Looped data into List.jsx and into Home.jsx

### PR DESCRIPTION
**Pull request by EmmaBin**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

Sarah and I worked on task #2. We updated _Home.jsx_ and _List.jsx_ to loop over the data and render the names of each shopping list a user is subscribed to, as well as the names of each item. One thing to note is that we haven’t inserted the **path** prop in _Home.jsx_, as we need further information to create the path. Since we don’t have a database created yet, we attempted to navigate to the /list page, except for one error from _firebase.js_, there are no errors caused by our implementation.

## Related Issue

2. As a user, I want to read all of my shopping lists and the items in them.

## Acceptance Criteria

- [x] In Home.jsx, the SingleList component is used to render the name of each shopping list a user is subscribed to
- [x] In List.jsx, the ListItem component is used to render the name of each item

## Type of Changes
`feature` 

### Before

DE3275B8-A3A8-41B6-A8F4-DABAA1AAE45B
1B739864-755E-483D-8FE9-5DAAAB9053B3

### After

Screenshot 2024-08-13 at 7 58 54 PM
Screenshot 2024-08-13 at 7 58 37 PM

### CODE REVIEW

**Comment by arandel1** [comment]
Looks good!

**Review by arandel1 — APPROVED** [review]
Looks good!

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
 			<p>
 				Hello from the home (<code>/</code>) page!
 			</p>
 			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `lists` array
-				 * so we can see which lists the user has access to.
-				 */}
+				{/* insert path as prop later */}
+				{data &&
```
Comment (mentalcaries) [inline src/views/Home.jsx:11]: Good job with including a null check here!

File: src/views/List.jsx [hunk 2]
```diff
@@ -7,6 +7,10 @@ export function List({ data }) {
 				Hello from the <code>/list</code> page!
 			</p>
 			<ul>
+				{data &&
+					data.map((list) => {
+						return <ListItem key={list.id} name={list.name} />;
```
Comment (mentalcaries) [inline src/views/List.jsx:12]: I like that you all passed the key prop here to the list item!

**Review by mentalcaries — APPROVED** [review]
A simple issue, but very well done!

File: src/views/Home.jsx [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
+					data.map((list) => {
+						return (
+							<SingleList
+								key={list.id}
+								name={list.name}
+								setListPath={setListPath}
```
Comment (mindyzwan) [inline src/views/Home.jsx:17]: Now that we have the right data structure in Firebase, go ahead and add the path here! This should work with that change! 👍

Comment (sar-mko) [reply src/views/Home.jsx]: Added it! Also, there was an error that the keys weren't unique so used the randomUUID() method to pass into the keys for now.

File: src/views/Home.jsx [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
-				 * so we can see which lists the user has access to.
-				 */}
+				{data &&
+					data.map((list) => {
+						return (
+							<SingleList
+								key={crypto.randomUUID()}
+								name={list.name}
+								setListPath={setListPath}
+								path={path}
```
Comment (mindyzwan) [inline src/views/Home.jsx:17]: This path actually comes from the list item itself!
```suggestion
								path={list.path}
```

File: src/views/Home.jsx [hunk 5]
```diff
@@ -1,16 +1,23 @@
 import './Home.css';
-
-export function Home({ data, setListPath }) {
+import { SingleList } from '../components';
+export function Home({ data, setListPath, path }) {
```
Comment (mindyzwan) [inline src/views/Home.jsx:3]: See other comment - the path comes from the list itself.

```suggestion
export function Home({ data, setListPath }) {
```

**Pull request merged**